src/VirtualClient/VirtualClient.Actions/MongoDB/MongoDBExecutor.cs

```diff
@@ -0,0 +1,235 @@
+            this.systemManager = this.Dependencies.GetService<ISystemManagement>();
+        }
+
+        /// <summary>
+        /// The retry policy to apply to package install for handling transient errors.
+        /// </summary>
+        public IAsyncPolicy InstallRetryPolicy { get; set; } = Policy
+            .Handle<WorkloadException>(exc => exc.Reason == ErrorReason.DependencyInstallationFailed)
+            .WaitAndRetryAsync(5, (retries) => TimeSpan.FromSeconds(retries * 2));
+
+        /// <summary>
+        /// Defines the name of the YCSB package.
+        /// </summary>
+        public string YCSBPackageName
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.YCSBPackageName));
+            }
+        }
+
+        /// <summary>
+        /// The file path where logs will be written.
+        /// </summary>
+        protected string YcsbPackagePath { get; set; }
+
+        /// <summary>
+        /// Workload package path.
+        /// </summary>
+        protected string MongoDBPackagePath { get; set; }
+
+        /// <summary>
+        /// Initializes the environment for execution of the MongoDB YCSB workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            DependencyPath mongoDBPackage = await this.packageManager.GetPackageAsync(this.PackageName, CancellationToken.None);
+
+            if (mongoDBPackage == null)
+            {
+                throw new DependencyException(
+                    $"The expected package '{this.PackageName}' does not exist on the system or is not registered.",
+                    ErrorReason.WorkloadDependencyMissing);
+            }
+
+            this.MongoDBPackagePath = mongoDBPackage.Path;
+
+            DependencyPath ycsbPackage = await this.packageManager.GetPackageAsync(this.YCSBPackageName, CancellationToken.None);
+
+            if (mongoDBPackage == null)
+            {
+                throw new DependencyException(
+                    $"The expected package '{this.PackageName}' does not exist on the system or is not registered.",
+                    ErrorReason.WorkloadDependencyMissing);
+            }
+
+            this.YcsbPackagePath = ycsbPackage.Path;
+
+            // string pathtobin = this.PlatformSpecifics.Combine(this.PackageDirectory, "bin");
+
+        }
+
+        /// <summary>
+        /// Executes the MongoDB YCSB workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string mountPath = this.PlatformSpecifics.Combine(this.MongoDBPackagePath, "mongodb-linux-x86_64-ubuntu2004-5.0.15", "bin", "mongod");
+
+            try
+            {
+                string ycsbPath = this.PlatformSpecifics.Combine(this.YcsbPackagePath, "ycsb-0.5.0", "bin", "ycsb");
```

Comment:
Please organize them properly with install, DB initialization, and then server/client process